core/src/main/scala/com/gu/tableversions/core/InMemoryTableVersions.scala

```diff
@@ -17,95 +13,63 @@ import com.gu.tableversions.core.util.RichRef._
 class InMemoryTableVersions[F[_]] private (allUpdates: Ref[F, TableUpdates])(implicit F: Sync[F])
     extends TableVersions[F] {
 
-  override def init(
-      table: TableName,
-      isSnapshot: Boolean,
-      userId: UserId,
-      message: UpdateMessage,
-      timestamp: Instant): F[Unit] =
-    allUpdates.update { prev =>
-      if (prev.contains(table)) prev
-      else
-        prev + (table -> List(
-          TableUpdate(
-            userId,
-            message,
-            timestamp = timestamp,
-            operations = List(InitTable(table, isSnapshot))
-          )))
+  override def commit(table: TableName, update: TableVersions.TableUpdate): F[Unit] = {
+    val applyUpdate: TableUpdates => Either[Exception, TableUpdates] = { currentTableUpdates =>
```

Comment:
how about match case style rather than if/else ?
```
case currentTableUpdates if currentTableUpdates contains table =>
        val updated = currentTableUpdates + (table -> TableState(
          currentVersion = update.header.id,
          updates = currentTableUpdates(table).updates :+ update))
        Right(updated)
case _ =>
        Left(unknownTableError(table))
```